scipy/linalg/tests/test_decomp.py

```diff
@@ -765,29 +769,79 @@ def test_eigh_tridiagonal(self):
                                       abs(self.evec[:, ind1:ind2+1]))
 
 
-def test_eigh():
-    DIM = 6
-    v = {'dim': (DIM,),
-         'dtype': ('f','d','F','D'),
-         'overwrite': (True, False),
-         'lower': (True, False),
-         'turbo': (True, False),
-         'eigvals': (None, (2, DIM-2))}
-
-    for dim in v['dim']:
-        for typ in v['dtype']:
-            for overwrite in v['overwrite']:
-                for turbo in v['turbo']:
-                    for eigenvalues in v['eigvals']:
-                        for lower in v['lower']:
-                            eigenhproblem_standard(
-                                   'ordinary',
-                                   dim, typ, overwrite, lower,
-                                   turbo, eigenvalues)
-                            eigenhproblem_general(
-                                   'general ',
-                                   dim, typ, overwrite, lower,
-                                   turbo, eigenvalues)
+class TestEigh:
+    def setup_class(self):
```

Comment:
It is actually doing that. Everytime the tests fires up it generates matrices and the proceeding class methods use that. It is a pytest feature that prepares the test class.

Reply:
Yes. Using the function I posted might make the code shorter and more readable. At the least, it would eliminate some repetition between preparing the `a` and `b` matrices.